src/mpi/datatype/typerep/src/typerep_yaksa_pack.c

```diff
@@ -139,6 +185,92 @@ static int typerep_do_pack(const void *inbuf, MPI_Aint incount, MPI_Datatype dat
     goto fn_exit;
 }
 
+/* This function checks whether the operation is supported in yaksa for the
+ * provided datatype */
+int MPIR_Typerep_reduce_is_supported(MPI_Op op, MPI_Datatype datatype)
+{
+    yaksa_op_t yaksa_op;
+
+    if (!MPIR_CVAR_ENABLE_YAKSA_REDUCTION)
+        return 0;
+
+    if ((datatype == MPI_DOUBLE || datatype == MPI_C_DOUBLE_COMPLEX) &&
+        (!MPIR_CVAR_GPU_DOUBLE_SUPPORT))
+        return 0;
+    if ((datatype == MPI_LONG_DOUBLE || datatype == MPI_C_LONG_DOUBLE_COMPLEX) &&
+        (!MPIR_CVAR_GPU_LONG_DOUBLE_SUPPORT))
+        return 0;
+
+    if ((datatype == MPI_C_COMPLEX || datatype == MPI_C_DOUBLE_COMPLEX ||
+         datatype == MPI_C_LONG_DOUBLE_COMPLEX) && (!MPIR_CVAR_YAKSA_COMPLEX_SUPPORT))
+        return 0;
+#ifdef HAVE_FORTRAN_BINDING
+    if ((datatype == MPI_COMPLEX || datatype == MPI_DOUBLE_COMPLEX || datatype == MPI_COMPLEX8 ||
+         datatype == MPI_COMPLEX16) && (!MPIR_CVAR_YAKSA_COMPLEX_SUPPORT))
+        return 0;
+    if (datatype == MPI_DOUBLE_COMPLEX && (!MPIR_CVAR_GPU_DOUBLE_SUPPORT))
+        return 0;
+#endif
+#ifdef HAVE_CXX_BINDING
+    if ((datatype == MPI_CXX_FLOAT_COMPLEX || datatype == MPI_CXX_DOUBLE_COMPLEX ||
+         datatype == MPI_CXX_LONG_DOUBLE_COMPLEX) && (!MPIR_CVAR_YAKSA_COMPLEX_SUPPORT))
+        return 0;
+    if (datatype == MPI_CXX_DOUBLE_COMPLEX && (!MPIR_CVAR_GPU_DOUBLE_SUPPORT))
+        return 0;
+    if (datatype == MPI_CXX_LONG_DOUBLE_COMPLEX && (!MPIR_CVAR_GPU_LONG_DOUBLE_SUPPORT))
+        return 0;
+#endif
+    if ((datatype == MPI_FLOAT || datatype == MPI_DOUBLE || datatype == MPI_LONG_DOUBLE) &&
+        (op == MPI_LXOR || op == MPI_LAND || op == MPI_LOR))
+        return 0;
+
+    yaksa_op = MPII_Typerep_get_yaksa_op(op);
+
+    if (yaksa_op == YAKSA_OP__LAST)
+        return 0;
+    return 1;
+}
+
+/* This function performs pack-reduction in yaksa. For contiguous datatypes no packing is needed */
+static int MPIR_Typerep_pack_reduce(const void *inbuf, MPI_Aint incount, MPI_Datatype datatype,
```

Comment:
We don't need `MPIR_` name space for static functions. If we, later on, decide to expose this function, we can add the name space back then.

Also, we don't really need `reduce` context here. How about just call it `op_pack` (and the other one `op_unpack`)?